Funcky/Extensions/EnumerableExtensions.cs

```diff
@@ -74,5 +74,77 @@ public static void Each<T>(this IEnumerable<T> elements, Action<T> action)
                 action(element);
             }
         }
+
+        /// <summary>
+        /// Returns the first element of a sequence as an option, or a None value if the sequence contains no elements.
+        /// </summary>
+        /// <typeparam name="TSource">the inner type of the enumerable.</typeparam>
+        public static Option<TSource> FirstOrNone<TSource>(this IEnumerable<TSource> source)
+            where TSource : notnull
+        {
+            return source.FirstOrDefault() is { } value
```

Comment:
`.FirstOrDefault` only works as intended for reference types

Reply:
=> New implementation with Select